pkg/kubectl/cmd/create/create_role.go

```diff
@@ -192,6 +192,11 @@ func (o *CreateRoleOptions) Complete(f cmdutil.Factory, cmd *cobra.Command, args
 		}
 		resource.Resource = parts[0]
 
+		if resource.Resource == "*" && len(parts) == 1 && len(sections) == 1 {
```

Comment:
Check this before splitting into sections

Reply:
Hmm, actually, we probably want to allow * within a specific group

Reply:
Yes, I think that this line is better to put here to support specific groups like `*.extensions`.

Reply:
did you want to add a test for `*.extensions`?

Reply:
Yes, I added.